plugins/pubIds/urn/URNPubIdPlugin.inc.php

```diff
@@ -0,0 +1,236 @@
+	 */
+	function getPubIdDisplayType() {
+		return 'URN';
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getPubIdFullName()
+	 */
+	function getPubIdFullName() {
+		return 'Uniform Resource Name';
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getResolvingURL()
+	 */
+	function getResolvingURL($contextId, $pubId) {
+		$resolverURL = $this->getSetting($contextId, 'urnResolver');
+		return $resolverURL . $pubId;
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getPubIdMetadataFile()
+	 */
+	function getPubIdMetadataFile() {
+		return $this->getTemplatePath().'urnSuffixEdit.tpl';
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getPubIdAssignFile()
+	 */
+	function getPubIdAssignFile() {
+		return $this->getTemplatePath().'urnAssign.tpl';
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::instantiateSettingsForm()
+	 */
+	function instantiateSettingsForm($contextId) {
+		$this->import('classes.form.URNSettingsForm');
+		return new URNSettingsForm($this, $contextId);
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getFormFieldNames()
+	 */
+	function getFormFieldNames() {
+		return array('urnSuffix');
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getAssignFormFieldName()
+	 */
+	function getAssignFormFieldName() {
+		return 'assignURN';
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getPrefixFieldName()
+	 */
+	function getPrefixFieldName() {
+		return 'urnPrefix';
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getSuffixFieldName()
+	 */
+	function getSuffixFieldName() {
+		return 'urnSuffix';
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getLinkActions()
+	 */
+	function getLinkActions($pubObject) {
+		$linkActions = array();
+		import('lib.pkp.classes.linkAction.request.RemoteActionConfirmationModal');
+		$application = PKPApplication::getApplication();
+		$request = $application->getRequest();
+		$userVars = $request->getUserVars();
+		$userVars['pubIdPlugIn'] = get_class($this);
+		// Clear object pub id
+		$linkActions['clearPubIdLinkActionURN'] = new LinkAction(
+			'clearPubId',
+			new RemoteActionConfirmationModal(
+				__('plugins.pubIds.urn.editor.clearObjectsURN.confirm'),
+				__('common.delete'),
+				$request->url(null, null, 'clearPubId', null, $userVars),
+				'modal_delete'
+			),
+			__('plugins.pubIds.urn.editor.clearObjectsURN'),
+			'delete',
+			__('plugins.pubIds.urn.editor.clearObjectsURN')
+		);
+		return $linkActions;
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getSuffixPatternsFieldNames()
+	 */
+	function getSuffixPatternsFieldNames() {
+		return  array(
+			'Submission' => 'urnSubmissionSuffixPattern',
+			'Representation' => 'urnRepresentationSuffixPattern',
+			'SubmissionFile' => 'urnSubmissionFileSuffixPattern',
+		);
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::getDAOFieldNames()
+	 */
+	function getDAOFieldNames() {
+		return array('pub-id::other::urn');
+	}
+
+	/**
+	 * @copydoc PKPPubIdPlugin::isObjectTypeEnabled()
+	 */
+	function isObjectTypeEnabled($pubObjectType, $contextId) {
+		return $this->getSetting($contextId, "enable${pubObjectType}URN") == '1';
```

Comment:
I'd prefer just `return (boolean) $this->getSetting($contextId, "enable${pubObjectType}URN")`. IIRC `getSetting` should return a boolean true, so comparing with a string-quoted 1 is convoluted.